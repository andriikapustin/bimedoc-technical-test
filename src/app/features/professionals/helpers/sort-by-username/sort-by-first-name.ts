import { IProfessional } from '../../../../models/professional.model';

export function sortByFirstName(professionals: IProfessional[]): IProfessional[] {
  return professionals.sort((a, b) => a.first_name.localeCompare(b.first_name));
}
