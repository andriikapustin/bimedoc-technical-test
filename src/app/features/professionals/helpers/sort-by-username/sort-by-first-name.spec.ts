import { IProfessional } from '../../../../models/professional.model';

import { sortByFirstName } from './sort-by-first-name';

describe(`sortProfessionalsByUserName()`, () => {
  const professionalsMock = [
    { first_name: 'Thomas' },
    { first_name: 'THOMAS' },
    { first_name: 'thomas' },
    { first_name: 'Andrii' },
    { first_name: 'Nicolas' },
    { first_name: 'Alex' },
  ] as IProfessional[];

  it('should return array', () => {
    expect(sortByFirstName(professionalsMock)).toBeInstanceOf(Array);
  });

  it('should return array with the same length', () => {
    expect(sortByFirstName(professionalsMock).length).toBe(professionalsMock.length);
  });

  it('should return correctly sorted array', () => {
    sortByFirstName(professionalsMock).forEach((current, i, array) => {
      const next = array[i + 1];
      const actual = current.first_name.localeCompare(next?.first_name);

      if (current.first_name === next?.first_name) {
        expect(actual).toBe(0);
      } else {
        expect(actual).toBe(-1);
      }
    });
  });

  it('should match snapshot', () => {
    expect(sortByFirstName(professionalsMock)).toMatchSnapshot();
  });
});
