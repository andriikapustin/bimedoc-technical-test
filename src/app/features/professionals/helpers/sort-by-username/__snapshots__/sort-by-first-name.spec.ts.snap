// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sortProfessionalsByUserName() should match snapshot 1`] = `
Array [
  Object {
    "first_name": "Alex",
  },
  Object {
    "first_name": "Andrii",
  },
  Object {
    "first_name": "Nicolas",
  },
  Object {
    "first_name": "thomas",
  },
  Object {
    "first_name": "Thomas",
  },
  Object {
    "first_name": "THOMAS",
  },
]
`;
